src/libraries/System.Runtime/tests/System.Runtime.Tests/System/Reflection/SignatureTypes.cs

```diff
@@ -237,6 +272,8 @@ public static void MakeSignatureArrayType()
             Assert.True(et.IsGenericParameter);
             Assert.False(et.IsGenericTypeParameter);
             Assert.True(et.IsGenericMethodParameter);
+            Assert.Throws<NotSupportedException>(() => et.IsValueType);
```

Comment:
This test method should be testing SignatureArrayType.

These should be `Assert.False(t.IsValueType)` and `Assert.False(t.IsEnum)` before `Type et = t.GetElementType();` line instead.

It is not very interesting to test `MakeGenericMethodParameter` here. It is tested in `MakeGenericMethodParameter` method.